ultra/ultra/env/ultra_env.py

```diff
@@ -58,7 +58,11 @@ def __init__(
         else:
             _scenarios = glob.glob(f"{self.scenarios['test']}")
 
-        self.ultra_scores = BaselineAdapter.reward_adapter
+        # NOTE: Does the ULTRA environment need an ultra_scores? Perhaps the reward
+        #       adapter should be agent-specific.
+        self.ultra_scores = adapters.adapter_from_type(
+            adapter_type=adapters.AdapterType.DefaultReward
+        )
```

Comment:
Right, okay I see that, it seems like though we are calculating a reward for the agent using the agent's reward adapter: https://github.com/huawei-noah/SMARTS/blob/2af0d66dca6dea72944baf7254e2e1a50534c210/ultra/ultra/env/rllib_ultra_env.py#L177 and then also calculating the reward again using `ultra_scores`: https://github.com/huawei-noah/SMARTS/blob/2af0d66dca6dea72944baf7254e2e1a50534c210/ultra/ultra/env/rllib_ultra_env.py#L130
I think I remember we had talked about this before, and that we chose to do it this way so that every agent, no matter what type of reward adapter they use, can compare rewards obtained from the same reward function. Maybe this is still the right way to go? We can maybe talk about it in the ULTRA meeting.

Reply:
Hmm, that seems to be correct, we want the ULTRA env reward to be agent independent

Reply:
Update: Having an ultra_scores is fine. The adapted reward is recorded under the "episode_return" key in our data for the experiment, and the ultra_scores in recorded under the "env_score" key in our data for the experiment.